app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -329,14 +357,22 @@ private void persistAdapterState(ListView view) {
             adapter = containerFragment.getData();
             // on orientation change
             if (adapter != null) {
-                view.setAdapter(adapter);
-                EntitySelectViewSetup.setupDivider(this, view, shortSelect.usesGridView());
-
-                findViewById(R.id.entity_select_loading).setVisibility(View.GONE);
+                setupUIFromAdapter(view);
             }
         }
     }
 
+    private void setupUIFromAdapter(ListView view) {
+        view.setAdapter(adapter);
+        EntitySelectViewSetup.setupDivider(this, view, shortSelect.usesGridView());
+
+        findViewById(R.id.entity_select_loading).setVisibility(View.GONE);
+
+        if (adapter.isFilteringByCalloutResult()) {

```

Comment:
I think that the text search query stuff should get restored by state restore logic that sets the search box text to `lastQueryString`. When that is set it calls the `afterTextChanged` which eventually triggers the search banner.

Regardless, I'm changing this to just call `setSearchBannerState`